compiler/ballerina-lang/src/main/java/io/ballerina/projects/NullBackend.java

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+*/
+package io.ballerina.projects;
+
+import io.ballerina.projects.environment.ProjectEnvironment;
+import io.ballerina.projects.internal.DefaultDiagnosticResult;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Ballerina compiler backend that does not emit backend code, used for testing BIR.
+ *
+ * @since 2.0
+ */
+public class NullBackend extends CompilerBackend {
```

Comment:
Do we need a backend here? The idea behind bancked is it take cares of compilation backend related tasks, such as codegen, interop validation, etc. But BIRGen is actually part of compiler frontend right? So having a Backend here does not make sense to me.

Reply:
This class was created because of this [comment](https://github.com/ballerina-platform/ballerina-lang/blob/d87f41f6f39e83424ddab7efe2d48b453a9a7c71/compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java#L76) wanting to make `PackageContext` an internal package. 
Would it be ok to access it from `BCompileUtil` class?

Reply:
You don't need the backend class here. Just calling pckCompilation.getCompilation would give you the symbol to access BIR

Reply:
I was wrong here. I was thinking codegen phase does not invoke birgen phase and was seperated out but looks like they are both part of generateCodeInternal method in ModuleContext class. Ideally, we should separate them out to different methods.

We have to go with separate backend class here as per the original PR.